types/react-router/index.d.ts

```diff
@@ -100,3 +101,4 @@ export interface match<P> {
 
 export function matchPath<P>(pathname: string, props: RouteProps): match<P> | null;
 export function withRouter<P>(component: React.ComponentType<RouteComponentProps<any> & P>): React.ComponentClass<P>;
+export function withRouter<P, TFunction extends React.ComponentClass<P>>(target: TFunction): TFunction; // decorator signature
```

Comment:
If we really are going to add this, write the comment in the line above, not in the end of the same one.
Otherwise people might miss it.